runtime/compiler/control/JITClientCompilationThread.cpp

```diff
@@ -3179,6 +3184,59 @@ remoteCompile(
          JITServer::ServerMemoryState nextMemoryState = std::get<9>(recv);
          updateCompThreadActivationPolicy(compInfoPT, nextMemoryState);
          }
+      else if (JITServer::MessageType::compilationThreadCrashed == response)
+         {
+         bool writeVerboseLog = TR::Options::isAnyVerboseOptionSet(
+               TR_VerboseJITServer,
+               TR_VerboseCompilationDispatch,
+               TR_VerbosePerformance,
+               TR_VerboseCompFailure
+               );
+         // A server compilation thread has crashed, temporarily stop compilations
+         // and request a JitDump recompilation of the method that caused the crash
+         if (writeVerboseLog)
+             TR_VerboseLog::writeLineLocked(TR_Vlog_FAILURE,
+               "Server compilation thread crashed while compiling %s (%s)", compiler->signature(), compiler->getHotnessName());
+         Trc_JITServerCompThreadCrashed(vmThread, compInfoPT->getCompThreadId(), compiler->signature(), compiler->getHotnessName());
+
+         compInfo->getPersistentInfo()->setDisableFurtherCompilation(true);
```

Comment:
Is `disableFurtherCompilation` needed on the client? Will this result in compilations failing for good and not being retried locally? The server should not accept new connection, so I am thinking that we can eliminate this line.

Reply:
We will deal with `disableFurtherCompilation` in `compileOnSeparateThread`. Compilations will be failed with `compilationSuspended` error code. Not sure if this means that compilations will not be retried. In my tests, I see tons of `compilationSuspended` failures once the server crashes, but the total number of compilations on the client is about the same as without a crash, so I assume they get requeued at some point. 
But yeah, just eliminating this line should be the simplest solution.